### PR TITLE
Add changelog

### PR DESCRIPTION
**Pull request by cptartur**
Closes #

## Introduced changes

- Added an initial changelog based on https://keepachangelog.com/en/1.1.0/

**Note I only included changes related to forge, cast and user-facing cheatable starknet changes will have to be included separately.**

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Changelog
```
Comment (THenry14) [inline CHANGELOG.md:1]: How should it look like? Do we separate it somehow to indicate what was done for which component, or is everything just put with each other?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+### Changed
+
+- `declare` return type to `starknet::ClassHash`, doesn't return a `Result`
+- `PreparedContract` `class_hash` changed to `starknet::ClassHash`
+- `deploy` return type to `starknet::ContractAddress`
+
+### Fixed
+
+- Using the same cairo file names as corelib files no longer fails test execution
+
```
Comment (Radinyn) [inline CHANGELOG.md:25]:
```suggestion
- All libfuncs are now allowed in contracts

```

#307

File: CHANGELOG.md (hunk 1 above)
Comment (cptartur) [reply CHANGELOG.md:1]: It depends on how we want our release notes to look like. I'm fine with including separate subheadins for forge and cast if we want to keep that separation.

Comment (MaksymilianDemitraszek) [reply CHANGELOG.md:1]: I think we should separate those, it will be cleaner

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
```
Comment (THenry14) [inline CHANGELOG.md:8]: I guess it's already released 🤷

**Review by Radinyn — APPROVED** [review]

**Pull request merged**